### PR TITLE
various fixes for Redis workbook

### PR DESCRIPTION
**Pull request by dougbrad**
Fix to show Overview if no Connections
Fix calculation of Rate
Fix CPU percentage (upon discussion w/ Auston, moved to VM Insights counter)
Add Memory percentage
Fix Command menu, various other fixes here
Fix calculation of call time
Various other fixes

* [x ] Explain your changes, so people looking at the PR know *what* and *why*, the code changes are the *how*.
* [x ] *if updating templates, it is helpful to post a screenshot of what the changes look like.*
* [ x] *if updating templates, ensure that your parameters and steps have meaningful names.*
* [ ] *if adding new templates, or changing items in a gallery, it's helpful to post a screenshot of what the gallery looks like now*

image

### CODE REVIEW

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook [hunk 1]
```diff
@@ -285,11 +282,11 @@
...
               "size": 0,
               "showAnalytics": true,
               "timeContext": {
-                "durationMs": 0
+                "durationMs": 14400000
```
Comment (acearun) [inline Workbooks/Workloads/Redis Insights/Redis Insights.workbook:289]: Its weird that the time context value changed even though it is tied to the TimeRange param in the line below.

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook [hunk 2]
```diff
@@ -397,19 +394,32 @@
...
                   },
                   {
-                    "columnMatch": "ProcessedTrend",
-                    "formatter": 9,
+                    "columnMatch": "MemoryUtilizationPercent",
+                    "formatter": 18,
                     "formatOptions": {
-                      "min": 0,
-                      "palette": "blue"
+                      "thresholdsOptions": "icons",
```
Comment (acearun) [inline Workbooks/Workloads/Redis Insights/Redis Insights.workbook:400]: Do we need a warning threshold too?

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook [hunk 3]
```diff
@@ -477,11 +491,11 @@
               "version": "KqlItem/1.0",
               "query": "InsightsMetrics\r\n| where (Computer in ({OnboardedComputers}) or '*' in ({OnboardedComputers})) \r\n| where Name == \"keyspace_hitrate\"\r\n| make-series Ratio = avg(Val) default=0 on TimeGenerated from {TimeRange:start} to {TimeRange:end} step {TimeRange:grain} by Computer",
               "size": 0,
-              "aggregation": 1,
+              "aggregation": 5,
               "showAnalytics": true,
-              "title": "Cache Hit Ratio",
+              "title": "Cache hit ratio",
```
Comment (acearun) [inline Workbooks/Workloads/Redis Insights/Redis Insights.workbook:496]: Didn't we change this to Cache hit ratio percentage in the alert?

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook [hunk 4]
```diff
@@ -669,10 +681,10 @@
               "size": 0,
               "aggregation": 3,
               "showAnalytics": true,
-              "title": "Connected Slaves",
+              "title": "Connected slaves",
```
Comment (acearun) [inline Workbooks/Workloads/Redis Insights/Redis Insights.workbook:684]: We should probably rename this to Connected replicas.

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook (hunk 1 above)
Comment (acearun) [reply Workbooks/Workloads/Redis Insights/Redis Insights.workbook:289]: @gardnerjr, do you know why this happened?

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook (hunk 2 above)
Comment (dougbrad) [reply Workbooks/Workloads/Redis Insights/Redis Insights.workbook:400]: Set as following for both CPU and Memory:
Warning: > 80%
Error: > 95%

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook (hunk 3 above)
Comment (dougbrad) [reply Workbooks/Workloads/Redis Insights/Redis Insights.workbook]: yes, needed to figure out the query so was going to defer, but yeah best to just do now - done
Need to update the alert query with this new learnings...

File: Workbooks/Workloads/Redis Insights/Redis Insights.workbook (hunk 4 above)
Comment (dougbrad) [reply Workbooks/Workloads/Redis Insights/Redis Insights.workbook]: done

**Pull request merged**